substrate/client/service/src/lib.rs

```diff
@@ -109,9 +109,9 @@ impl RpcHandlers {
 	pub async fn rpc_query(
 		&self,
 		json_query: &str,
-	) -> Result<(String, mpsc::UnboundedReceiver<String>), JsonRpseeError> {
+	) -> Result<(String, tokio::sync::mpsc::Receiver<String>), JsonRpseeError> {
 		self.0
-			.raw_json_request(json_query)
+			.raw_json_request(json_query, tokio::sync::Semaphore::MAX_PERMITS)
```

Comment:
Why are we passing `tokio::sync::Semaphore::MAX_PERMITS` as channel size here?

Reply:
Ideally we want to use `usize::MAX` but because `tokio::sync::mpsc::channel` panics then I changed it to `MAX_PERMITS` as upperbound and added a comment about it in the code.